### PR TITLE
Type hints / mypy errors

### PR DESCRIPTION
**Pull request by to-sta**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR adds type annotations to the python backend. It solves the current errors from running mypy.

I ran mypy till there were no errors anymore:
`Success: no issues found in 42 source files`

Additional: i ran the `docker-compose up --build` to check if build works as expected.

### Related issue

- #329

### CODE REVIEW

File: backend/authentication/admin.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from .models import User
 
 
-class UserAdmin(admin.ModelAdmin):
+class UserAdmin(admin.ModelAdmin[User]):
```
Comment (andrewtavis) [inline backend/authentication/admin.py:6]: I deleted some of the `[Class]` arguments in the merge conflict update as it looks like they weren't used in other places in the PR for #295. Let's keep this in mind and add them back in if necessary. Sorry, just trying to get through all of these and see the new state of things, but we can do a clean up at the end 😊

File: backend/authentication/models.py [hunk 2]
```diff
@@ -1,89 +1,89 @@
...
+        "SupportEntityType", on_delete=models.CASCADE, related_name="supporter"
+    )
+    supporter_entity = models.IntegerField(null=True)
+    supported_type = models.ForeignKey(
+        "SupportEntityType", on_delete=models.CASCADE, related_name="supported"
+    )
+    supported_entity = models.IntegerField(null=True)
+
+    def __str__(self) -> str:
+        return str(self.id)
```
Comment (andrewtavis) [inline backend/authentication/models.py:44]: We need `str()` here as we have this as a numeric field, so this is necessary here but isn't for `return self.username`?

File: backend/authentication/models.py [hunk 3]
```diff
@@ -1,89 +1,89 @@
...
+    verification_method = models.CharField(max_length=30, null=True)
+    verification_partner = models.ForeignKey(
+        "User", on_delete=models.SET_NULL, null=True, blank=True
+    )
+    social_accounts = ArrayField(models.CharField(max_length=255), null=True)
+    total_flags = models.IntegerField(default=0)
+    creation_date = models.DateTimeField(auto_now_add=True)
+    deletion_date = models.DateField(null=True)
+
+    def __str__(self) ->str:
```
Comment (andrewtavis) [inline backend/authentication/models.py:64]: Space needed (I'll get it).

File: backend/authentication/admin.py (hunk 1 above)
Comment (andrewtavis) [reply backend/authentication/admin.py:6]: These were added back in in ef1c8a8 :)

File: backend/authentication/models.py (hunk 3 above)
Comment (andrewtavis) [reply backend/authentication/models.py:64]: Fixed in ef1c8a8 :)

File: backend/authentication/models.py (hunk 2 above)
Comment (to-sta) [reply backend/authentication/models.py:44]: @andrewtavis exactly! It's necessary for all non string fields (IntegerField, DateField,...). The __str__ method is special case where the `return` datatype is already determined to be a `str`. What we specifiy as return value will be shown in the admin view (`localhost:8000/v1/admin/`).

**Pull request merged**